File: shared/formatting.py
"""Uniform response shape for every tool across every provider."""

from __future__ import annotations

from typing import Any


def ok(data: Any, *, provider: str, model: str, usage: dict | None = None) -> dict:
    return {
        "ok": True,
        "provider": provider,
        "model": model,
        "usage": usage or {},
        "data": data,
    }

File: src/shared/formatting.ts
export interface OkResponse {
  ok: true;
  provider: string;
  model: string;
  usage: Record<string, unknown>;
  data: unknown;
}

export function ok(
  data: unknown,
  opts: { provider: string; model: string; usage?: Record<string, unknown> | null },
): OkResponse {
  return {
    ok: true,
    provider: opts.provider,
    model: opts.model,
    usage: opts.usage ?? {},
    data,
  };
}
